kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go

```diff
@@ -115,7 +115,9 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		kafkachannelLister:   kafkaChannelInformer.Lister(),
 		kafkachannelInformer: kafkaChannelInformer.Informer(),
 	}
-	r.impl = kafkachannelreconciler.NewImpl(ctx, r)
+	r.impl = kafkachannelreconciler.NewImpl(ctx, r, func(impl *controller.Impl) controller.Options {
```

Comment:
I think this is not going to work because the reconciler updates the status here:
https://github.com/knative/eventing-contrib/blob/master/kafka/channel/pkg/reconciler/dispatcher/kafkachannel.go#L172

Reply:
Yeah, status updates need to be moved to the controller, see also https://github.com/knative-sandbox/eventing-kafka/issues/98.

Reply:
ah, yeah - let's do it on the new repo

Reply:
@matzew I think on the new repo the issue is the same. kafka ch dispatcher always updates that status